app/models/reports/schema_v2.rb

```diff
@@ -137,6 +138,23 @@ def missed_visits_with_ltfu_rates
       missed_visits_rates(with_ltfu: true)
     end
 
+    memoize def visited_without_bp_taken(with_ltfu: false)
+      if with_ltfu
+        regions.each_with_object({}) { |region, hsh| hsh[region.slug] = sum(region, :adjusted_visited_no_bp_lost_to_follow_up) }
+      else
+        regions.each_with_object({}) { |region, hsh| hsh[region.slug] = sum(region, :adjusted_visited_no_bp_under_care) }
+      end
+    end
+
+    memoize def visited_without_bp_taken_rates(with_ltfu: false)
+      region_period_cached_query(__method__, with_ltfu: with_ltfu) do |entry|
```

Comment:
Same question [as in previous PR](https://github.com/simpledotorg/simple-server/pull/2767/files#r684976196): Is this is for convenience, or do we need caching here?

Reply:
I think we will want some caching, when I compare this to v1 code its slightly slower for very large production like data sets.  But I think we can revisit performance a bit later (see https://app.clubhouse.io/simpledotorg/story/4516/benchmark-reports-comparing-v1-to-v2).